### PR TITLE
fix: remove parse5 types

### PR DESCRIPTION
**Pull request by SimenB**
Please fill in this template.

If removing a declaration:
- [ ] If a package was never on Definitely Typed, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [x] Delete the package's directory.
- [x] Add it to `notNeededPackages.json`.

https://github.com/inikulin/parse5/releases/tag/v7.0.0

### CODE REVIEW

**Comment by SimenB** [comment]
Hah, a billion errors. I won't get to them, but maybe one of the maintainers want to?

**Comment by SimenB** [comment]
Good point, pushed commits removing the others as well 👍

**Comment by SimenB** [comment]
> with not-needed, stuff that uses the old version should still work right? v7 has many, many breaking changes

Might need to explicitly depend on currently published `@types/parse5*` packages, I don't know DT well enough to tell

**Comment by SimenB** [comment]
Actually, there is no v7 of https://www.npmjs.com/package/parse5-serializer-stream, so reverted that

**Comment by SimenB** [comment]
@43081j CI passed, so seems to work! 👍

File: types/xmlserializer/package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
     "private": true,
     "dependencies": {
-        "parse5": "^5.1.0"
```
Comment (SimenB) [inline types/xmlserializer/package.json:4]: no idea why it depended on "normal" `parse5`, v7 is the first with types

**Comment by 43081j** [comment]
maybe @sandersn or someone can give us some guidance here just to be sure we're doing it right.

looks good to me but it would be good to understand whats going on with that xmlserializer dependency, and to know if we did the right thing depending on a types package in the manifest

File: types/xmlserializer/package.json (hunk 1 above)
Comment (sandersn) [reply types/xmlserializer/package.json:4]: that must have been a mistake; I think the new code is correct

**Review by sandersn — CHANGES_REQUESTED** [review]
The change is correct *if* the types for jsdom and parse5-serializer-stream shouldn't be updated to parse5@7. Since I believe jsdom, at least, ships its own types, then it doesn't make sense to update the DT types.

Is the same true for parse5-serialiser-stream? If so, then this change is ready to go.

**Review by sandersn — APPROVED** [review]

**Comment by SimenB** [comment]
Ready to merge

File: types/jsdom/package.json [hunk 2]
```diff
@@ -1,5 +1,8 @@
 {
     "private": true,
+    "dependencies": {
+        "@types/parse5": "^6.0.3"
```
[the remark this answers is not in the document]
Comment (SimenB) [reply types/jsdom/package.json:4]: yeah, migrating to using `parse5` directly is unblocked by this change, see #61171 (although I think that ended up working out)

[the remark this answers is not in the document]
Comment (hasezoey) [reply types/jsdom/package.json:4]: > though I think jsdom@20 ships its own types so I don't think it's possible in this case

no jsdom still does not ship its own types (at least from what i can tell)

**Pull request merged**